lesson17-18/home_work17-18.md

```diff
@@ -0,0 +1,274 @@
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [S], cksum 0x4101 (correct), seq 3790498554, win 64240, options [mss 1460,nop,wscale 8,nop,nop,sackOK], length 0
+09:01:51.902827 IP (tos 0x0, ttl 128, id 366, offset 0, flags [DF], proto TCP (6), length 40)
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [.], cksum 0xd79a (correct), seq 3790498555, ack 1797265894, win 8212, length 0
+09:01:51.908962 IP (tos 0x0, ttl 128, id 367, offset 0, flags [DF], proto TCP (6), length 68)
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [P.], cksum 0xe165 (correct), seq 0:28, ack 1, win 8212, length 28
+09:01:51.917908 IP (tos 0x0, ttl 128, id 368, offset 0, flags [DF], proto TCP (6), length 1296)
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [P.], cksum 0x8423 (correct), seq 28:1284, ack 22, win 8212, length 1256
+09:01:51.920325 IP (tos 0x0, ttl 128, id 369, offset 0, flags [DF], proto TCP (6), length 88)
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [P.], cksum 0x3f5e (correct), seq 1284:1332, ack 1302, win 8207, length 48
+09:01:51.933171 IP (tos 0x0, ttl 128, id 370, offset 0, flags [DF], proto TCP (6), length 120)
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [P.], cksum 0x2191 (correct), seq 1332:1412, ack 1606, win 8212, length 80
+09:01:51.973922 IP (tos 0x0, ttl 128, id 371, offset 0, flags [DF], proto TCP (6), length 40)
+    192.168.56.1.62565 > localhost.localdomain.ssh: Flags [.], cksum 0xcb91 (correct), seq 1412, ack 1670, win 8212, length 0
+```
+
+#### 3. Close session. Find in tcpdump output packets that are related to TCP session closure.
+```bash
+09:03:32.163325 IP (tos 0x0, ttl 128, id 417, offset 0, flags [DF], proto TCP (6), length 104)
+    192.168.56.1.62599 > localhost.localdomain.ssh: Flags [P.], cksum 0x51dd (correct), seq 2388:2452, ack 3142, win 8212, length 64
+09:03:32.164690 IP (tos 0x0, ttl 128, id 418, offset 0, flags [DF], proto TCP (6), length 40)
+    192.168.56.1.62599 > localhost.localdomain.ssh: Flags [.], cksum 0x9b4a (correct), seq 2452, ack 3382, win 8211, length 0
+09:03:32.167863 IP (tos 0x0, ttl 128, id 419, offset 0, flags [DF], proto TCP (6), length 40)
+    192.168.56.1.62599 > localhost.localdomain.ssh: Flags [F.], cksum 0x9b49 (correct), seq 2452, ack 3382, win 8211, length 0
+09:03:32.168360 IP (tos 0x0, ttl 128, id 420, offset 0, flags [DF], proto TCP (6), length 40)
+    192.168.56.1.62599 > localhost.localdomain.ssh: Flags [.], cksum 0x9b48 (correct), seq 2453, ack 3383, win 8211, length 0
+```
+
+#### 4. run tcpdump and request any http site in separate session. Find HTTP request and answer packets with ASCII data in it.  Tcpdump command must be as strict as possible to capture only needed packages for this http request.
+> Enother screen `curl http://nginx.org/`
+```bash
+[sit2@localhost ~]$ sudo tcpdump -nn -A "(port 80 and host 10.0.2.15)"
```

Comment:
The tcpdump filter could be more "strict". You can add something more (for now you have only port and host)

Reply:
@Cakemanator I can add an interface
`[sit2@localhost ~]$ sudo tcpdump -nn -A -i enp0s3  "(port 80 and host 10.0.2.15)"`

Reply:
@Cakemanator I tried to do it
`[sit2@localhost ~]$ sudo tcpdump -nn -A -i enp0s3 "(dst  port 80 and src 10.0.2.15)"`

Reply:
What if you'll specify protocol? Port 80 is usually for http, but we don't know that it was not reconfigured on the destination side.

Reply:
like this.
`[sit2@localhost ~]$ sudo tcpdump -A -i enp0s3 "src 10.0.2.15 and tcp port http"`

Reply:
Do we need to specify port at all? 
We're trying to sniff traffic to `http://nginx.org`, we know protocol, url address, but don't know (at least for sure) the port.

Reply:
@Cakemanator I cannot find in the list of http protocols in tcpdump. Without protocols it looks like this
` sudo tcpdump -A -i enp0s3 src 10.0.2.15`

Reply:
@Cakemanator  I think I can use tcp protocols like this , since http uses tcp to connect
`[sit2@localhost ~]$ sudo tcpdump -A -i enp0s3 "src 10.0.2.15 and ip proto \tcp"`

Reply:
Ok
And what if we want to get incoming traffic as well? Will this command take it?

Reply:
@Cakemanator In this case, we need to specify the following.
`[sit2@localhost ~]$ sudo tcpdump -A -i enp0s3 "host 10.0.2.15 and ip proto \tcp"`

Reply:
Ok

Reply:
@Cakemanator can I merge ?

Reply:
I have one more question for line 238.

Reply:
You wrote the same sentence 2 times, didn't you? Above ?

Reply:
Pardon, my bad :)
Add the discussed changes and you can merge after that